modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheProxy.java

```diff
@@ -780,6 +780,13 @@ private ClusterGroup projection(boolean loc) {
 
                 final SqlQuery p = (SqlQuery)qry;
 
+                if (ctx.isReplicated() && p.getPartitions() != null)
```

Comment:
We should not look at `ctx.isReplicated()` here anymore because we are allowed to run any type of query from any cache (partitioned or replicated). This check probably must be somewhere else.